src/ngraph/runtime/intelgpu/intelgpu_backend.cpp

```diff
@@ -230,7 +260,9 @@ bool runtime::intelgpu::IntelGPUBackend::compile(shared_ptr<Function> func)
 
     for (shared_ptr<Node> op : func->get_ops())
     {
-        if ("Parameter" == op->description())
+        switch (get_typeid(op->description()))
```

Comment:
You should add the gcc pragmas
```
#pragma GCC diagnostic push
#pragma GCC diagnostic error "-Wswitch"
#pragma GCC diagnostic error "-Wswitch-enum"
```
So that all cases are required. For unsupported ops you would simply throw an unsupported_op exception inside the case. No default: is need for this as all cases are covered.